renku/core/management/migrations/m_0003__2_initial.py

```diff
@@ -93,27 +92,30 @@ def _migrate_datasets_pre_v0_3(client):
 def _migrate_broken_dataset_paths(client):
     """Ensure all paths are using correct directory structure."""
     for dataset in get_client_datasets(client):
-        expected_path = client.renku_datasets_path / dataset.identifier
         if not dataset.name:
             dataset.name = generate_default_name(dataset.title, dataset.version)
 
         # migrate the refs
-        ref = LinkReference.create(client=client, name="datasets/{0}".format(dataset.name), force=True,)
+        ref = LinkReference.create(client=client, name="datasets/{0}".format(dataset.name), force=True)
```

Comment:
This can cause an exception because `LinkReference.check_ref_format` is called to validate the name of the reference, but dataset names can contain characters not allowed in the ref format.